File: src/main/java/org/trp/shincolle/entity/base/MountAiNumbers.java
package org.trp.shincolle.entity.base;

final class MountAiNumbers {
    static final double SHIP_FLOATING_DEPTH = 0.3D;
    static final double BUOY_MIN_DEPTH = 0.15D;
    static final double BUOY_COEFF = 0.035D;
    static final double BUOY_EXPONENT = 0.6D;
    static final double BUOY_OFFSET = 0.005D;
    static final double BUOY_DAMP = 0.80D;
    static final double BUOY_MAX_MOTION = 0.1D;

    static final int FOLLOW_TELEPORT_COOLDOWN_TICKS = 100;
    static final int FOLLOW_STUCK_TICK_LIMIT = 120;
    static final double FOLLOW_TELEPORT_DISTANCE_SQ = 256.0D;
    static final double FOLLOW_MOVE_SPEED = 1.0D;
    static final double FOLLOW_WIDTH_PADDING = 0.75D;
    static final double TELEPORT_VERTICAL_OFFSET = 0.75D;

    static final float LOOK_YAW = 30.0F;
    static final float LOOK_PITCH = 30.0F;
    static final int FOLLOW_BEHAVIOR_PRIORITY = 2;
    static final int RANGE_ATTACK_BEHAVIOR_PRIORITY = 11;
    static final int RANDOM_STROLL_BEHAVIOR_PRIORITY = 25;
    static final double RANGE_ATTACK_MOVE_SPEED = 0.8D;
    static final double RANDOM_STROLL_SPEED = 0.8D;
    static final int RANDOM_STROLL_CHANCE = 120;
    static final int RANDOM_STROLL_HORIZONTAL_RANGE = 10;
    static final int RANDOM_STROLL_VERTICAL_RANGE = 7;

    static final int STOP_SHIP_AI_AIR_SUPPLY = 300;
    static final int SERVER_SYNC_INTERVAL_MASK = 0x1F;
    static final int AIR_SUPPLY_INTERVAL_MASK = 0x7F;
    static final double RIDER_INTERACT_DISTANCE_SQ = 16.0D;
    static final double ROTATION_EPSILON = 0.001D;
    static final double SUBMARINE_DISABLE_DEPTH = 0.4D;
    static final double PARTICLE_MOTION_LIMIT = 0.25D;
    static final int PARTICLE_BASE_AMOUNT = 2;
    static final int PARTICLE_RANDOM_AMOUNT = 3;
    static final double PARTICLE_TRAIL_SCALE = 3.0D;
    static final double PARTICLE_WIDTH_CENTER_OFFSET = 0.5D;
    static final double PARTICLE_WIDTH_Y_SCALE = 0.15D;
    static final double PARTICLE_Y_OFFSET = 0.6D;
    static final double PARTICLE_SPEED_SCALE = 1.5D;
    static final int WATER_PARTICLE_INTERVAL = 2;
    static final float FORWARD_BRAKE_FACTOR = 0.25F;
    static final float STRAFE_FACTOR = 0.5F;
    static final float SUBMARINE_PITCH_THRESHOLD = 60.0F;

    static final double HOST_MAX_HEALTH_SCALE = 0.5D;
    static final double FOLLOW_RANGE_ATTR = 64.0D;
    static final double BASE_MAX_HEALTH = 4.0D;
    static final double BASE_MOVEMENT_SPEED = 0.3D;
    static final double BASE_KNOCKBACK_RESISTANCE = 1.0D;
    static final double BASE_STEP_HEIGHT = 3.0D;
    static final double BASE_ATTACK_DAMAGE = 2.0D;

    static final int AIM_BASE_TICKS = 10;
    static final float AIM_SCALE_TICKS = 20.0F;
    static final int LEVEL_CAP = 150;
    static final double MIN_ATTACK_RANGE = 1.0D;

    private MountAiNumbers() {
    }
}

File: src/main/java/org/trp/shincolle/entity/base/MountAiNumbers.kt
package org.trp.shincolle.entity.base

internal object MountAiNumbers {
    const val SHIP_FLOATING_DEPTH: Double = 0.3
    const val BUOY_MIN_DEPTH: Double = 0.15
    const val BUOY_COEFF: Double = 0.035
    const val BUOY_EXPONENT: Double = 0.6
    const val BUOY_OFFSET: Double = 0.005
    const val BUOY_DAMP: Double = 0.80
    const val BUOY_MAX_MOTION: Double = 0.1

    const val FOLLOW_TELEPORT_COOLDOWN_TICKS: Int = 100
    const val FOLLOW_STUCK_TICK_LIMIT: Int = 120
    const val FOLLOW_TELEPORT_DISTANCE_SQ: Double = 256.0
    const val FOLLOW_MOVE_SPEED: Double = 1.0
    const val FOLLOW_WIDTH_PADDING: Double = 0.75
    const val TELEPORT_VERTICAL_OFFSET: Double = 0.75

    const val LOOK_YAW: Float = 30.0f
    const val LOOK_PITCH: Float = 30.0f
    const val FOLLOW_BEHAVIOR_PRIORITY: Int = 2
    const val RANGE_ATTACK_BEHAVIOR_PRIORITY: Int = 11
    const val RANDOM_STROLL_BEHAVIOR_PRIORITY: Int = 25
    const val RANGE_ATTACK_MOVE_SPEED: Double = 0.8
    const val RANDOM_STROLL_SPEED: Double = 0.8
    const val RANDOM_STROLL_CHANCE: Int = 120
    const val RANDOM_STROLL_HORIZONTAL_RANGE: Int = 10
    const val RANDOM_STROLL_VERTICAL_RANGE: Int = 7

    const val STOP_SHIP_AI_AIR_SUPPLY: Int = 300
    const val SERVER_SYNC_INTERVAL_MASK: Int = 0x1F
    const val AIR_SUPPLY_INTERVAL_MASK: Int = 0x7F
    const val RIDER_INTERACT_DISTANCE_SQ: Double = 16.0
    const val ROTATION_EPSILON: Double = 0.001
    const val SUBMARINE_DISABLE_DEPTH: Double = 0.4
    const val PARTICLE_MOTION_LIMIT: Double = 0.25
    const val PARTICLE_BASE_AMOUNT: Int = 2
    const val PARTICLE_RANDOM_AMOUNT: Int = 3
    const val PARTICLE_TRAIL_SCALE: Double = 3.0
    const val PARTICLE_WIDTH_CENTER_OFFSET: Double = 0.5
    const val PARTICLE_WIDTH_Y_SCALE: Double = 0.15
    const val PARTICLE_Y_OFFSET: Double = 0.6
    const val PARTICLE_SPEED_SCALE: Double = 1.5
    const val WATER_PARTICLE_INTERVAL: Int = 2
    const val FORWARD_BRAKE_FACTOR: Float = 0.25f
    const val STRAFE_FACTOR: Float = 0.5f
    const val SUBMARINE_PITCH_THRESHOLD: Float = 60.0f

    const val HOST_MAX_HEALTH_SCALE: Double = 0.5
    const val FOLLOW_RANGE_ATTR: Double = 64.0
    const val BASE_MAX_HEALTH: Double = 4.0
    const val BASE_MOVEMENT_SPEED: Double = 0.3
    const val BASE_KNOCKBACK_RESISTANCE: Double = 1.0
    const val BASE_STEP_HEIGHT: Double = 3.0
    const val BASE_ATTACK_DAMAGE: Double = 2.0

    const val AIM_BASE_TICKS: Int = 10
    const val AIM_SCALE_TICKS: Float = 20.0f
    const val LEVEL_CAP: Int = 150
    const val MIN_ATTACK_RANGE: Double = 1.0
}
